finality-aleph/src/new_network/session.rs

```diff
@@ -0,0 +1,115 @@
+use crate::{
+    crypto::{AuthorityPen, AuthorityVerifier},
+    new_network::{ComponentNetwork, SendError, Sender as SenderTrait, SessionCommand},
+    NodeIndex, SessionId,
+};
+use aleph_bft::Recipient;
+use codec::Codec;
+use futures::channel::mpsc;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+
+/// Sends data within a single session.
+#[derive(Clone)]
+pub struct Sender<D: Clone + Codec + Send> {
+    session_id: SessionId,
+    messages_for_network: mpsc::UnboundedSender<(D, SessionId, Recipient)>,
+}
+
+impl<D: Clone + Codec + Send> SenderTrait<D> for Sender<D> {
+    fn send(&self, data: D, recipient: Recipient) -> Result<(), SendError> {
+        self.messages_for_network
+            .unbounded_send((data, self.session_id, recipient))
+            .map_err(|_| SendError::SendFailed)
+    }
+}
+
+/// Sends and receives data within a single session.
+pub struct Network<D: Clone + Codec + Send> {
+    sender: Sender<D>,
+    receiver: Arc<Mutex<mpsc::UnboundedReceiver<D>>>,
+}
+
+impl<D: Clone + Codec + Send> ComponentNetwork<D> for Network<D> {
```

Comment:
What value brings having 3 different Networks: struct Network, trait ComponentNetwork and trait DataNetwork? Are those additional layers really worth an additional complexity? Who do we expect to implement those traits on different types than we are using right now? Kind of YAGNI for me.

Reply:
mickey_mouse_a_surprize_that_will_help_us_later.apng

To actually answer the question – the next PR will introduce a way of splitting networks generically, which is needed to create two interfaces, one for `aleph_bft::Network` and another for the justification component. This way has to use the components, thus the need for a `ComponentNetwork`, same for `Sender` and `Receiver`.